### ISSUE TITLE
[Feature] Enable building of responsive apps 

### ISSUE THREAD

**Issue by areyabhishek** [appsmithorg/appsmith#4630]
## Summary
We need to allow users to build or modify pages for different viewports. The complete scope needs to be written here. Just creating this as a placeholder.

**Comment by somangshu** [comment]
Related issue > #1043

**Comment by abhvsn** [comment]
A user on discord requested a feature which might be related to this epic  https://github.com/appsmithorg/appsmith/issues/11154#

**Comment by cverdes** [comment]
Perhaps a possible alternative is tu be able to define different coordinates and sizes for the in page widgets for different canvas sizes.
Or let the coordinates and sizes of each widget be defined by JavaScript formulas using the current size of the canva.
Just ideas...

**Comment by somangshu** [comment]
@cverdes thanks for the inputs, With the first point, Did you mean we could introduce breakpoints and develop on them separately, That the easiest fix we too are thinking of right now. Let me know if you mean otherwise

**Comment by cverdes** [comment]
The idea is to do something similar to how Microsoft Power App handles responsive layouts.
The scripts have access to `App.Width` and `App.Height` to know the canvas size and `Parent.Width` and `Parent.Height` in case the widgets is within a container.
With this information, and letting to set, by scripting, the values of `x`, `y`, `width` and `height` of each widget, the user can compute the location an size any widget in any place depending of the canvas size and the cointaner where it is in.
Perhaps [this link](https://docs.microsoft.com/en-us/powerapps/maker/canvas-apps/create-responsive-layout ) from Microsoft can help to complement my explanation.
It's an easy and quick to implement idea that I think can improve the responsive layouts in Appsmith.
Best Regards.

**Comment by billyrudi** [comment]
Hi, there are news about development of this request?
it is very important that it is carried out so that you can best use appsmith on industrial terminals in warehouses for example

**Comment by Nikhil-Nandagopal** [comment]
Hi @billyrudi we haven't begun on this feature yet. You can build an app for a tablet / mobile screen size today but making it responsive on all devices needs some thought from our end

**Comment by Nikhil-Nandagopal** [comment]
@billyrudi @cverdes @pierredewilde Hey we're now researching this feature. Would either of you be interested in doing a user interview with us about this? You can find a time on my calendar below if you'd like to participate
https://calendly.com/appsmith-nikhil/30min?back=1&month=2022-05

**Comment by Nikhil-Nandagopal** [comment]
@lucasa that is because it's an epic epic :)
We'd love some feedback from you on what we're building. If you're available for a quick chat, please block some time on my calendar https://calendly.com/appsmith-nikhil/appsmith-meeting?back=1&month=2022-11

**Comment by yasharma2301** [comment]
Hey @Nikhil-Nandagopal any updates on this feature's timeline? Dropping a conversation I had on Discord community regarding the same [here](https://discord.com/channels/725602949748752515/1043158104231133265)

**Comment by Nikhil-Nandagopal** [comment]
@yasharma2301 yes we do have some update on this. I'd love for you to block some time on my calendar to get some feedback https://calendly.com/appsmith-nikhil/appsmith-meeting

**Comment by mweichert** [comment]
Eager to learn more about this feature as well!

**Comment by ame-appsmith** [comment]
[Another user](https://app.frontapp.com/open/msg_1mlyzmq4?key=Xs42lOokWnlGqNWdM_hxfqyeRsLW6guD) is interested in this feature.

**Comment by jsandona** [comment]
I am also interested in this feature.

We have a form for our employees to submit invoices. Some use it from their desktops and others need access from their mobile devices while on the road. Currently my app works well on desktop using the fluid width, but on mobile all widgets are shrunk and become unusable.

**Comment by Nikhil-Nandagopal** [comment]
@jsandona we'd be happy to give you an early preview of the feature to get your feedback. You can block some time on my calendar below
https://calendly.com/appsmith-nikhil/appsmith-meeting

**Comment by josh9383** [comment]
I would like to contribute to it, let me know how

**Comment by BrianJM** [comment]
> @jsandona we'd be happy to give you an early preview of the feature to get your feedback. You can block some time on my calendar below
> https://calendly.com/appsmith-nikhil/appsmith-meeting

I'd like to participate in the early preview program. Please let me know if this is possible.

**Comment by rohitagarwal88** [comment]
Yes, @BrianJM. Please block some time on Nikhil's calendar https://calendly.com/appsmith-nikhil/appsmith-meeting for the early preview.

**Comment by richardwhatever** [comment]
+1 this please!

**Comment by cripsisxyz** [comment]
+1 this :)

**Comment by andreevanatasha** [comment]
Hey all @pierredewilde @cverdes @billyrudi @lucasa @yasharma2301 @mweichert @jsandona @josh9383 @BrianJM @richardwhatever @cripsisxyz, thank you for your interest in mobile responsiveness! I'm a product manager with the team that owns Appsmith's UI Builder (that's where you drag and drop and move widgets around!), so this epic is our responsibility. 

To make building responsive apps truly efficient, we want to go beyond implementing  individual features and dive deeper into our users’ needs and context. 

That’s why I would love to learn more about your experience of designing and building interfaces (not necessarily in Appsmith). What constitutes a great app building experience? How an app interface is designed, put together, tested and maintained? What are the things that you hate in React/Bootstrap/CSS/any other frontend technology, and would love to finally see done right?

If you have built or worked with custom software in your company, automated a workflow, created a user interface for database management, or streamlined a process in any other way — it doesn’t have to be an Appsmith app! — and are willing to talk about that experience, please book us a slot: https://calendly.com/natasha-lko/ui-building 

Real-world experience and insights will shape the future of UI building in Appsmith — we cannot do it without you 🙏
